api-analysis.md

```diff
@@ -0,0 +1,137 @@
+GET  /api/v2/participants/npq
+GET  /api/v2/participants/npq/{participant_id}
+PUT  /api/v2/participants/npq/{participant_id}/change-schedule
+PUT  /api/v2/participants/npq/{participant_id}/defer
+PUT  /api/v2/participants/npq/{participant_id}/resume
+PUT  /api/v2/participants/npq/{participant_id}/withdraw
+GET  /api/v2/participants/npq/outcomes
+GET  /api/v2/participants/npq/{participant_id}/outcomes
+POST /api/v2/participants/npq/{participant_id}/outcomes
+GET /api/v3/participant-declarations
+```
+
+### V3
+
+```
+GET  /api/v3/npq-applications
+GET  /api/v3/npq-applications/{application_id}
+POST /api/v3/npq-applications/{application_id}/accept
+POST /api/v3/npq-applications/{application_id}/reject
+GET  /api/v3/participants/npq/outcomes
+GET  /api/v3/participants/npq/{participant_id}/outcomes
+POST /api/v3/participants/npq/{participant_id}/outcomes
+GET  /api/v3/participants/npq
+GET  /api/v3/participants/npq/{participant_id}
+PUT  /api/v3/participants/npq/{participant_id}/change-schedule
+PUT  /api/v3/participants/npq/{participant_id}/defer
+PUT  /api/v3/participants/npq/{participant_id}/resume
+PUT  /api/v3/participants/npq/{participant_id}/withdraw
+GET /api/v3/participant-declarations
+```
+
+### Identifiers: technical analysis
+
+The following analysis applies to the endpoints of the different API versions: v1, v2 and v3.
+
+#### API endpoints that receive an `application_id`
+
+```
+# samples
+GET  /api/v3/npq-applications/{application_id}
+POST /api/v3/npq-applications/{application_id}/accept
+POST /api/v3/npq-applications/{application_id}/reject
+```
+
+The parameter `application_id`:
+
+1. Is generated by ECF.
+2. Is [stored in the ￼`ecf_idf`￼](https://github.com/DFE-Digital/npq-registration/blob/452b3831254eb32c3e72c6118ee2980a1df72b6f/db/schema.rb#L32) in NPQ
+3. Is returned to Lead Providers via [API endpoint response](%5Bresponse%20of%20this%20endpoint%5D%28https%3A//manage-training-for-early-career-teachers.education.gov.uk/api-reference/reference-v3.html#api-v3-npq-applications-id-accept-post-parameters%29).
+
+##### Actions
+
+1. No need to migrate ECF `application_id` as this information is already present in the attribute: `ecf_id` in the `applications` table in NPQ.
+2. `application_id` will be returned to LP via the [response of this endpoint](https://manage-training-for-early-career-teachers.education.gov.uk/api-reference/reference-v3.html#api-v3-npq-applications-id-accept-post-parameters):
+   1. For existing applications (previous to the split), [it will return ￼`ecf_id`￼ from the current ￼`applications`￼ table](https://github.com/DFE-Digital/npq-registration/blob/98cdef7b2e879588499a2cd4d3e72a087782c415/db/schema.rb#L33).
+   2. For new applications (after split), it will return a new `UUID` which will be also stored in the same `applications` table (`ecf_id`) field.
+3. Rename `ecf_id` to `application_lead_provider_id`, so we break the link between NPQ and ECF (ECF is no longer the owner of the identifiers).
```

Comment:
I wonder if we could/should always keep the ones we generate vs the ones from ECF separate, to avoid confusion and possible misalignment, as I reckon they'll be a bit of that in the early days of separation, and to help with the audit and debugging?

Reply:
discussed in person: adding a new field to capture source or maybe a timestamp, other option for ids is too complex for lookups and API